server/bot_api.go

```diff
@@ -54,6 +59,184 @@ func (p *Plugin) handleGetUserImage(w http.ResponseWriter, r *http.Request, user
+	if err != nil {
+		res.Err = err.Error()
+		appErr, ok := err.(*model.AppError)
+		if ok {
+			res.Code = appErr.StatusCode
+		} else {
+			res.Code = http.StatusInternalServerError
+		}
+		return
+	}
+
+	// Upload is incomplete.
+	if fi == nil {
+		res.Code = http.StatusNoContent
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(fi); err != nil {
+		p.LogError(err.Error())
+	}
+}
+
+func (p *Plugin) handleBotCreateUpload(w http.ResponseWriter, r *http.Request) {
+	var res httpResponse
+	defer p.httpAudit("handleBotCreateUpload", &res, w, r)
+
+	var us *model.UploadSession
+	if err := json.NewDecoder(http.MaxBytesReader(w, r.Body, requestBodyMaxSizeBytes)).Decode(&us); err != nil {
+		res.Err = "failed to decode request body: " + err.Error()
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	us.Id = model.NewId()
+	us.Type = model.UploadTypeAttachment
+	us.UserId = p.getBotID()
+
+	us, err := p.API.CreateUploadSession(us)
+	if err != nil {
+		res.Err = err.Error()
+		appErr, ok := err.(*model.AppError)
+		if ok {
+			res.Code = appErr.StatusCode
+		} else {
+			res.Code = http.StatusInternalServerError
+		}
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(us); err != nil {
+		p.LogError(err.Error())
+	}
+}
+
+func (p *Plugin) handleBotPostRecordings(w http.ResponseWriter, r *http.Request, callID string) {
+	var res httpResponse
+	defer p.httpAudit("handleBotPostRecordings", &res, w, r)
+
+	var info map[string]string
+	if err := json.NewDecoder(http.MaxBytesReader(w, r.Body, requestBodyMaxSizeBytes)).Decode(&info); err != nil {
+		res.Err = "failed to decode request body: " + err.Error()
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	threadID := info["thread_id"]
+	if threadID == "" {
+		res.Err = "missing thread_id from request body"
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	fileID := info["file_id"]
+	if fileID == "" {
+		res.Err = "missing file_id from request body"
+		res.Code = http.StatusBadRequest
+		return
+	}
+
+	// Update call thread
+	post, appErr := p.API.GetPost(threadID)
+	if appErr != nil {
+		res.Err = "failed to get call thread: " + appErr.Error()
+		res.Code = http.StatusInternalServerError
+		return
+	}
+	recordings, ok := post.GetProp("recording_files").([]string)
+	if !ok {
+		recordings = []string{
+			fileID,
+		}
+	} else {
+		recordings = append(recordings, fileID)
+	}
+	post.AddProp("recording_files", recordings)
+	_, appErr = p.API.UpdatePost(post)
+	if appErr != nil {
+		res.Err = "failed to update call thread: " + appErr.Error()
+		res.Code = http.StatusInternalServerError
+		return
+	}
+
+	startAt, _ := post.GetProp("start_at").(int64)
+	postMsg := "Here's the call recording"
+	if title := post.GetProp("title"); title != "" {
+		postMsg = fmt.Sprintf("%s of %s at %s UTC", postMsg, title, time.UnixMilli(startAt).Format("3:04PM"))
+	}
```

Comment:
I think this should only serve as a fallback but likely most of it should be overridden by the custom post component on the front-end.